### PR TITLE
proper makefile with install and uninstall

### PR DESCRIPTION
**Pull request by veltzer**
I did some work on the project. Here is the summary:
- created a real makefile with install and uninstall targets configured for a modern ubuntu system
- fixed up compiler warnings (which were correct)
- tested this on ubuntu 17.10 (works)

### CODE REVIEW

File: performance.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+It is better to have the names in ~/.hosts in the:
```
Comment (bAndie91) [inline performance.txt:1]: could you explain?

File: TODO.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+- make bash completions complete host names from ~/.hosts
```
Comment (bAndie91) [inline TODO.txt:1]: could you please maintain the TODO items in github issue tracker rather than this file?

File: libnss_homehosts.c [hunk 3]
```diff
@@ -116,7 +119,7 @@ enum nss_status homehosts_gethostent_r(
 	char homehosts_file[PATH_MAX+1];
 	char ipbuf[INET6_ADDRSTRLEN+1];
 	char namebuf[_POSIX_HOST_NAME_MAX+1];
-	char ffmt_ip[7];	// fscanf format string
+	char ffmt_ip[10];	// fscanf format string
```
Comment (bAndie91) [inline libnss_homehosts.c:122]: 👍

**Comment by bAndie91** [comment]
i really appreciate your contribution, @veltzer . sorry for noticing it this late. please feel free to ping me when i'm seeming idle.

File: performance.txt (hunk 1 above)
Comment (veltzer) [reply performance.txt:1]: If you have the names with the domain in the ~/.hosts file then ubuntu 17 does not make a seperate domain search (at least that's how it's configured) and just gives back the ip. That was the result of my test.

File: TODO.txt (hunk 2 above)
Comment (veltzer) [reply TODO.txt:1]: Sure. I'll do that..

File: libnss_homehosts.c (hunk 3 above)
Comment (veltzer) [reply libnss_homehosts.c:122]: This, as I recall, was what caused the segfault...

File: performance.txt (hunk 1 above)
Comment (bAndie91) [reply performance.txt:1]: then it's only slightly related to libnss_homehosts. i'm good to refer to this finding here, but with indication that it's not related to this lib, rather related to libresolv, plus maybe some reference to `resolv.conf(5)` (especially to `options`/`ndots` I suspect your finding is related to).

File: libnss_homehosts.c (hunk 3 above)
Comment (bAndie91) [reply libnss_homehosts.c:122]: yes, certainly. i underallocated this one.

File: performance.txt (hunk 1 above)
Comment (veltzer) [reply performance.txt:1]: Your call. The reason I mentioned this is that if someone uses libnss_homehosts and then sees that the performance is bad he/she will connect that finding to libnss_homehosts and not to other factors. That is what happened to me until I understood it is not this libraries fault...

Comment (bAndie91) [reply performance.txt:1]: good point. let's include your performance advice with more explanation referring to libresolv.

**Comment by veltzer** [comment]
Can you integrate what you like and leave the rest for me?

File: libnss_homehosts.c (hunk 3 above)
Comment (bAndie91) [reply libnss_homehosts.c:122]: resolve #1

**Pull request merged**